src/app/common/hooks/use-background-location-redirect.ts

```diff
@@ -0,0 +1,34 @@
+import { useEffect } from 'react';
```

Comment:
I added this hook so that:
- If you have a modal open
- Open a new tab
- It will update the `backgroundLocation` to be the home page

https://github.com/hirosystems/wallet/assets/2938440/1be88221-e4d9-494f-bcaf-fe5d6159fb8f

Without doing this, an empty screen appears behind the modal

Reply:
Thinking on this further: couldn't each modal component handle their own `backgroundLocation` logic?

Where the sign out modal component has a check in its render that, if there's no `backgroundLocation`, it should re-router to having a default value of home?

Could work with a similar hook API, but where all the logic is colocated to the modal itself, rather than one big catch-all. 

Something like:

```ts
useDefaultBackgroundLocation(RouteUrls.Home)
```

which internally handles the `if (!backgroundLocation) navigate(currentRoute, { state: { backgroundLocation: defaultVal } })`

Reply:
Yes we can definitely do that. I just implemented a catch all for now as `Home` seems like the logical background. I can change it to accept a preferred default route

I can give it a try but I think once you open a new tab the `location.state` gets wiped completely.